sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/VertxHttpClient.java

```diff
@@ -188,14 +190,18 @@ private void sendBody(ContextView contextView, HttpRequest azureRequest, Progres
                 || bodyContent instanceof ByteBufferContent
                 || bodyContent instanceof StringContent
                 || bodyContent instanceof SerializableContent) {
+                // This cannot produce a NullPointerException for the BinaryDataContent types that trigger this as they
+                // are in-memory and have a length.
                 long contentLength = bodyContent.getLength();
-                vertxRequest.send(Buffer.buffer(Unpooled.wrappedBuffer(bodyContent.toByteBuffer())))
+                vertxRequest.send(Buffer.buffer(FluxUtil.byteBufferToArray(bodyContent.toByteBuffer())))
```

Comment:
why not 

```suggestion
                vertxRequest.send(Buffer.buffer(bodyContent.toBytes()))
```

?

(diff rendering is acting out)

Reply:
Good catch, will make that change